client/repoutils.c

```diff
@@ -39,6 +39,105 @@ TDNFRepoMakeCacheDirs(
     goto cleanup;
 }
 
+uint32_t
+TDNFRepoGetMetalink(
+    PTDNF pTdnf,
+    const char* pszRepo,
+    char** ppszMetalink
+    )
+{
+
+    uint32_t dwError = 0;
+    char* pszMetalink = NULL;
+    PTDNF_REPO_DATA_INTERNAL pRepos = NULL;
+
+    if(!pTdnf || IsNullOrEmptyString(pszRepo) || !ppszMetalink)
+    {
+        dwError = ERROR_TDNF_INVALID_PARAMETER;
+        BAIL_ON_TDNF_ERROR(dwError);
+    }
+    if(!pTdnf->pRepos)
+    {
+        dwError = ERROR_TDNF_NO_REPOS;
+        BAIL_ON_TDNF_ERROR(dwError);
+    }
+    pRepos = pTdnf->pRepos;
+
+    while(pRepos)
+    {
+        if(!strcmp(pszRepo, pRepos->pszId))
+        {
+            break;
+        }
+        pRepos = pRepos->pNext;
+    }
+
+    if(!pRepos)
+    {
+        dwError = ERROR_TDNF_REPO_NOT_FOUND;
+        BAIL_ON_TDNF_ERROR(dwError);
+    }
+
+    dwError = TDNFAllocateString(pRepos->pszMetaLink, &pszMetalink);
+    BAIL_ON_TDNF_ERROR(dwError);
+
+    *ppszMetalink = pszMetalink;
+
+cleanup:
+    return dwError;
+
+error:
+    TDNF_SAFE_FREE_MEMORY(pszMetalink);
+    goto cleanup;
+
+}
+uint32_t
+TDNFRepoSetBaseUrl(
+    PTDNF pTdnf,
+    const char* pszRepo,
+    char* ppszBaseUrl
```

Comment:
Is this a double pointer? Naming says ppszBaseUrl but looks to be a char pointer.

Reply:
Yeah, better to rename. Thanks!